app/pages/settings/profile.spec.js

```diff
@@ -0,0 +1,51 @@
+import React from 'react';
+import assert from 'assert';
+import { shallow } from 'enzyme';
+import CustomiseProfile from './profile';
+
+const resources = {
+  avatar: {
+    src: '//zooniverse.org/images/avatar.jpg',
+    delete() {}
+  },
+  profile_header: {
+    src: '//zooniverse.org/images/header.jpg',
+    delete() {}
+  }
+};
+
+const user = {
+  get(type) {
+    return Promise.resolve([resources[type]]);
+  }
+};
+
+describe('CustomiseProfile', () => {
+  let wrapper;
+
+  before(() => {
+    wrapper = shallow(<CustomiseProfile user={user} />);
+  });
+
+  beforeEach(() => wrapper.update());
+
+  it('renders the user avatar', () => {
+    const avatar = wrapper.find('ImageSelector[src="//zooniverse.org/images/avatar.jpg"]');
+    assert.equal(avatar.length, 1);
+  });
+
+  it('renders the user background', () => {
+    const avatar = wrapper.find('ImageSelector[src="//zooniverse.org/images/header.jpg"]');
+    assert.equal(avatar.length, 1);
+  });
+
+  it('deletes the avatar when Clear Avatar is pressed', () => {
+    wrapper.find('button').first().simulate('click');
+    setTimeout(() => assert.equal(wrapper.update().state().avatar, {}));
```

Comment:
@eatyourgreens Is this intended to simulate the avatar resource being an empty object when deleted? If so, shouldn't the test fail if I add`src: 'avatar'` inside `{}`?

Reply:
Yep.

Reply:
It doesn't.

Reply:
Interesting. I'll take another look at this when I'm free. Did you find any clues as to why it isn't working?

Reply:
I've never seen `setTimeout` used in test so I was checking if it plays well with `wrapper.update()`.